Automattic/wp-calypso
# Seo Preview: Use siteHasFeature selector

#### Changes proposed in this Pull Request

* Replaces `hasSiteSeoFeature` with a `siteHasFeature` call.
* Removes now unused functions and tests.

#### Testing instructions



* Go to your list of drafts https://container-recursing-panini.calypso.live/posts/drafts
* Click the ellipses menu and select "Preview"
* On the top left of the preview modal, select "Search & Social".
* On sites with plans below Business it should show an upgrade prompt <img width="1510" alt="Screen Shot 2022-05-10 at 2 05 33 PM" src="https://user-images.githubusercontent.com/1398304/167721931-209cf010-d6e1-4ea1-98fd-b1418105656f.png">

* On higher-tier sites it should show the preview. <img width="1510" alt="Screen Shot 2022-05-10 at 2 06 41 PM" src="https://user-images.githubusercontent.com/1398304/167721942-563c4fa9-be4e-43f9-b221-978910b22a45.png">




Related to p4TIVU-a66-p2
